3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/IgluAdapter.scala

```diff
@@ -47,10 +48,45 @@ object IgluAdapter extends Adapter {
 
   // Tracker version for an Iglu-compatible webhook
   private val TrackerVersion = "com.snowplowanalytics.iglu-v1"
+  private val VendorName = "Iglu"
 
   // Create a simple formatter function
   private val IgluFormatter: FormatterFunc = buildFormatter() // For defaults
 
+    /**
+     *
+     * Converts a payload into a single validated event
+     * Expects a valid json - returns a single failure if one is not present
+     *
+     * @param body json payload as POST'd by a webhook
+     * @param payload the rest of the payload details
+     * @param schemaUri the schemaUri for the event
+     * @param params The query string parameters
+     * @return a single validated event
+     */
+    private def payloadBodyToEvent(body: String, payload: CollectorPayload, schemaUri: String, params: Map[String, String]): Validated[RawEvent] = {
+      val parsed = parseJsonSafe(body) match {
+        case Success(jsonBody) => jsonBody
```

Comment:
If the parse fails we need to match for Failure as well otherwise this will throw an exception.